webapp/src/modules/auction/sagas.js

```diff
@@ -71,3 +90,49 @@ function* handleBidRequest(action) {
     yield put(bidOnParcelsFailure(error.message))
   }
 }
+
+function* handleFetchAuctionRateRequest(action) {
+  const { token } = action
+  const selectedToken = yield select(state => getSelectedToken(state))
+  if (selectedToken !== token) {
+    yield put(navigateTo(locations.auction(token)))
+  }
+  try {
+    const { currentPrice } = yield select(getParams)
+    const landAuction = eth.getContract('LANDAuction')
+
+    let rate = 1
+    if (TOKENS.MANA !== TOKENS[selectedToken]) {
+      rate = eth.utils.fromWei(
+        yield call(() =>
+          landAuction.getRate(
+            TOKENS.MANA,
+            TOKENS[selectedToken],
+            eth.utils.toWei(currentPrice)
+          )
+        )
+      )
+    }
+    yield put(fetchAuctionRateSuccess(selectedToken, rate))
+  } catch (error) {
+    yield put(fetchAuctionRateFailure(selectedToken, error.message))
+  }
+}
+
+function* handleFetchAuctionRateSuccess(action) {
+  const token = yield select(state => getSelectedToken(state))
+  yield put(fetchAuctionRateRequest(token))
+}
+
+function* handleConnectWalletSuccess(action) {
+  let connected = true
+  // keep refreshing params and rate while the user is on /auction
+  while (connected) {
+    const pathname = yield select(state => getPathname(state))
+    if (pathname === locations.auction()) {
+      yield put(fetchAuctionParamsRequest())
+    }
+    yield delay(REFRESH_INTERVAL)
+    connected = yield select(state => isConnected(state))
```

Comment:
Same as above with the `select` calls

This is really cool btw, I like checking the location from here

Reply:
✅ done